packages/@aws-cdk/cdk/lib/synthesis.ts

```diff
@@ -74,14 +73,16 @@ export class SynthesisSession implements ISynthesisSession {
     return 'synthesize' in obj;
   }
 
-  public readonly store: ISessionStore;
+  public readonly outdir: string;
 
   private readonly artifacts: { [id: string]: cxapi.Artifact } = { };
-  private readonly buildSteps: { [id: string]: cxapi.BuildStep } = { };
   private _manifest?: cxapi.AssemblyManifest;
 
-  constructor(options: SynthesisOptions) {
-    this.store = options.store || new InMemoryStore();
+  constructor(options: SynthesisOptions = { }) {
+    this.outdir = options.outdir || fs.mkdtempSync(path.join(os.tmpdir(), 'cdk.out'));
+    if (!fs.existsSync(this.outdir)) {
```

Comment:
On the contrary, if the location exists:
- an error should be raised if it is not a directory
- an error should be raised if the directory is not empty (unless some `--force` is passeD?)

Reply:
We actually utilize this directory as a cache for assets, so if an asset with a certain source hash was already staged, we avoid the expensive copying, so I actually rely on the fact that this directory is long-lived.

I added a check that this is a directory if it exists.